### ISSUE TITLE
Serverless function to retrieve Project

### ISSUE THREAD

**Issue by replygirl** [BSA-US/Dual-Power-App-Client#10]
**Goal:** Create a serverless function that retrieves, cleans up, and returns the contents of a GitHub Project

**Use case:** When viewing #1 or an “about” page, contents of the [Roadmap Project](https://github.com/BSA-US/Dual-Power-App-Client/projects/6) can be retrieved from a serverless function for display

**Data we know we’ll need**
For each item:
- Title
- Body
- URL
- Column
For each column:
- Title
- Order

**Caveats:**
- There is no design solution in place
- Data listed above may not capture everything we need
- Interacting with Projects through the GitHub APIs may be restricted to the [GitHub Developer Program](https://developer.github.com/program/) & thus may require an auth key in our environment variables. If so, we should discuss how Now Secrets will be named & documented.

**Helpful resources:**
- API v3: [Projects](https://developer.github.com/v3/projects/)
- API v4 (GraphQL):
  - [Project](https://developer.github.com/v4/object/project/)
  - [Organization](https://developer.github.com/v4/object/organization/)

**Comment by alexoxford** [comment]
What is the purpose of this exactly? Is a detailed, developer-oriented roadmap really needed on a user-facing page? Would a simple overview of the roadmap (or even just a screenshot) be enough?

**Comment by odbol** [comment]
I agree that this might be a bit overkilll. I like Alex's screenshot idea, although it's not accessible. But perhaps just a link to the projects roadmap page, or a bit more fancy: an iframe with it embedded in the page, would be sufficient.

**Comment by alexoxford** [comment]
Ah, good call on the accessibility!
